Bugsnag/Configuration/BugsnagConfiguration.m

```diff
@@ -455,16 +450,23 @@ -(void)deletePersistedUserData {
 // MARK: - Properties: Getters and Setters
 // -----------------------------------------------------------------------------
 
+@synthesize maxBreadcrumbs = _maxBreadcrumbs;
+
 - (NSUInteger)maxBreadcrumbs {
-    return self.breadcrumbs.capacity;
+    @synchronized (self) {
```

Comment:
I don't see why this needs to take out a lock - I'm aware this is done elsewhere for other properties but think this could be removed for the majority if not all of those custom accessors, unless there's a specific reason. Possibly something to raise in another ticket.

Reply:
I'd rather stick with the pattern used already - will raise a ticket to consider this in future.